staging/src/k8s.io/apimachinery/pkg/api/validate/schema_test.go

```diff
@@ -0,0 +1,1017 @@
+			value := 0
+			return ForbiddenPointer(c, fp, &value, nil)
+		},
+		err: "fldpath: Forbidden",
+	}, {
+		fn: func(c operation.Context, fp *field.Path) field.ErrorList {
+			pointer := (*bool)(nil)
+			return ForbiddenPointer(c, fp, pointer, nil)
+		},
+	}, {
+		fn: func(c operation.Context, fp *field.Path) field.ErrorList {
+			value := false
+			return ForbiddenPointer(c, fp, &value, nil)
+		},
+		err: "fldpath: Forbidden",
+	}, {
+		fn: func(c operation.Context, fp *field.Path) field.ErrorList {
+			pointer := (*struct{ S string })(nil)
+			return ForbiddenPointer(c, fp, pointer, nil)
+		},
+	}, {
+		fn: func(c operation.Context, fp *field.Path) field.ErrorList {
+			value := struct{ S string }{}
+			return ForbiddenPointer(c, fp, &value, nil)
+		},
+		err: "fldpath: Forbidden",
+	}, {
+		fn: func(c operation.Context, fp *field.Path) field.ErrorList {
+			pointer := (**string)(nil)
+			return ForbiddenPointer(c, fp, pointer, nil)
+		},
+	}, {
+		fn: func(c operation.Context, fp *field.Path) field.ErrorList {
+			value := (*string)(nil)
+			return ForbiddenPointer(c, fp, &value, nil)
+		},
+		err: "fldpath: Forbidden",
+	}}
+
+	for i, tc := range cases {
+		result := tc.fn(operation.Context{}, field.NewPath("fldpath"))
+		if len(result) > 0 && tc.err == "" {
+			t.Errorf("case %d: unexpected failure: %v", i, fmtErrs(result))
+			continue
+		}
+		if len(result) == 0 && tc.err != "" {
+			t.Errorf("case %d: unexpected success: expected %q", i, tc.err)
+			continue
+		}
+		if len(result) > 0 {
+			if len(result) > 1 {
+				t.Errorf("case %d: unexepected multi-error: %v", i, fmtErrs(result))
+				continue
+			}
+			if re := regexp.MustCompile(tc.err); !re.MatchString(result[0].Error()) {
+				t.Errorf("case %d: wrong error\nexpected: %q\n     got: %v", i, tc.err, fmtErrs(result))
+			}
+		}
+	}
+}
+
+func TestForbiddenSlice(t *testing.T) {
+	cases := []struct {
+		fn  func(c operation.Context, fp *field.Path) field.ErrorList
+		err string // regex
+	}{{
+		fn: func(c operation.Context, fp *field.Path) field.ErrorList {
+			value := []string{}
+			return ForbiddenSlice(c, fp, value, nil)
+		},
+	}, {
+		fn: func(c operation.Context, fp *field.Path) field.ErrorList {
+			value := []string{""}
+			return ForbiddenSlice(c, fp, value, nil)
+		},
+		err: "fldpath: Forbidden",
+	}, {
+		fn: func(c operation.Context, fp *field.Path) field.ErrorList {
+			value := []int{}
+			return ForbiddenSlice(c, fp, value, nil)
+		},
+	}, {
+		fn: func(c operation.Context, fp *field.Path) field.ErrorList {
+			value := []int{0}
+			return ForbiddenSlice(c, fp, value, nil)
+		},
+		err: "fldpath: Forbidden",
+	}, {
+		fn: func(c operation.Context, fp *field.Path) field.ErrorList {
+			value := []bool{}
+			return ForbiddenSlice(c, fp, value, nil)
+		},
+	}, {
+		fn: func(c operation.Context, fp *field.Path) field.ErrorList {
+			value := []bool{false}
+			return ForbiddenSlice(c, fp, value, nil)
+		},
+		err: "fldpath: Forbidden",
+	}}
```

Comment:
How about adding a test of a slice of pointers?

Reply:
sure, done.